src/lib/class/specialized_string/Password/methods.cpp

```diff
@@ -0,0 +1,65 @@
+/* ************************************************************************** */
+/*                                                                            */
+/*                                                        :::      ::::::::   */
+/*   methods.cpp                                        :+:      :+:    :+:   */
+/*                                                    +:+ +:+         +:+     */
+/*   By: gle-mini <gle-mini@student.42.fr>          +#+  +:+       +#+        */
+/*                                                +#+#+#+#+#+   +#+           */
+/*   Created: 2024/03/04 22:37:23 by jodufour          #+#    #+#             */
+/*   Updated: 2024/03/14 16:58:36 by gle-mini         ###   ########.fr       */
+/*                                                                            */
+/* ************************************************************************** */
+
+#include "abnf_components.hpp"
+#include "class/specialized_string/Password.hpp"
+
+#define MAXIMUM_LENGTH_FOR_PASSWORD 50
+
+inline static bool is_nolfcrspcl(char const c)
+{
+	return std::string("\x00\r\n :").find(c) == std::string::npos;
+}
+
+inline static bool is_nolfcrsp(char const c)
+{
+	return std::string("\x00\r\n ").find(c) == std::string::npos;
```

Comment:
As I replied in the thread of my previous comment, you won't ever be able to match NUL characters using the std::string::find method, so please, remove it from the characters to match.